GalnetMonitor/GalnetMonitor.cs

```diff
@@ -232,126 +233,82 @@ public void HandleProfile(JObject profile)
-                    return "Community Goal";
-                }
+                return GetGalnetResource("categoryPowerplay");
+            }
 
-                if (title == "Galactic News: Starport Status Update")
-                {
-                    return "Starport Status Update";
-                }
+            if (title.StartsWith(GetGalnetResource("titleFilterCg")) ||
+                Regex.IsMatch(content, GetGalnetResource("contentFilterCgRegex")))
+            {
+                return GetGalnetResource("categoryCG");
+            }
 
-                return "Article";
+            if (title.StartsWith(GetGalnetResource("titleFilterStarportStatus")))
+            {
+                return GetGalnetResource("categoryStarportStatus");
             }
-            else if (configuration.language == "Français")
+
+            if (title.StartsWith(GetGalnetResource("titleFilterWeekInReview")))
             {
-                if (title.StartsWith("Actualité galactique : Rapport hebdomadaire - "))
-                {
-                    string subtitle = title.Replace("Actualité galactique : Rapport hebdomadaire - ", "");
-                    if (subtitle == "Démocratie")
-                    {
-                        return "Democracy Report";
-                    }
-                    else if (subtitle == "Conflits")
-                    {
-                        return "Conflict Report";
-                    }
-                    else if (subtitle == "Santé")
-                    {
-                        return "Health Report";
-                    }
-                    else if (subtitle == "Économie")
-                    {
-                        return "Economic Report";
-                    }
-                    else if (subtitle == "Sécurité")
-                    {
-                        return "Security Report";
-                    }
-                    else if (subtitle == "Expansions")
-                    {
-                        return "Expansion Report";
-                    }
-                }
+                return GetGalnetResource("categoryWeekInReview");
+            }
 
-                if (title.StartsWith("Opération communautaire"))
-                {
-                    return "Community Goal";
-                }
+            return GetGalnetResource("categoryArticle");
+        }
 
-                if (title == "Actualité galactique : Mise à jour - État des spatioports")
-                {
-                    return "Starport Status Update";
-                }
+        private string GetGalnetResource(string basename)
+        {
+            CultureInfo ci = locale != null ? CultureInfo.GetCultureInfo(locale) : CultureInfo.InvariantCulture;
+            return resourceManager.GetString(basename, ci) ?? null;
+        }
 
-                return "Article";
-            }
-            else if (configuration.language == "Deutsch")
+        public Dictionary<string, string> GetGalnetLocales()
```

Comment:
There is some duplication with the main app's code here. In general, the "Don't Repeat Yourself" rule is important for ease of maintenance but I'm going to let this slide as I sense the GalnetLocales and the main app locales may need to diverge.

Reply:
Oh, I know it rhymes. I considered moving the applicable code to a Utilities.cs where it could be accessible to both. But as you already noted the code does diverge a bit.